ts/src/bybit.ts

```diff
@@ -3496,18 +3496,18 @@ export default class bybit extends Exchange {
         const isTakeProfitTriggerOrder = takeProfitTriggerPrice !== undefined;
         const isStopLoss = stopLoss !== undefined;
         const isTakeProfit = takeProfit !== undefined;
+        const isBuy = side === 'buy';
+        const ascending = stopLossTriggerPrice ? !isBuy : isBuy;
         if (triggerPrice !== undefined) {
-            const isBuy = side === 'buy';
-            const ascending = stopLossTriggerPrice ? !isBuy : isBuy;
             request['triggerDirection'] = ascending ? 2 : 1;
             request['triggerPrice'] = this.priceToPrecision (symbol, triggerPrice);
         } else if (isStopLossTriggerOrder || isTakeProfitTriggerOrder) {
             if (isStopLossTriggerOrder) {
```

Comment:
can we refactor this if clause

Reply:
seems like it's a bit unnecessary, the code could be

```
request['triggerDirection'] = ascending ? 2 : 1;
const triggerPrice = isStopLossTriggerOrder ? stopLossTriggerPrice : takeProfitTriggerPrice
request['triggerPrice'] = this.priceToPrecision (symbol, triggerPrice);
```

Reply:
I changed this slightly but left the conditional so trigger orders can be used to open a position but stopLossTrigger and takeProfitTrigger orders are reduceOnly